deepcell_spots/applications/spot_detection.py

```diff
@@ -0,0 +1,252 @@
+                                 self.required_rank, len(image.shape)))
+
+        if image.shape[-1] != self.required_channels:
+            raise ValueError('Input data must have {} channels. '
+                             'Input data only has {} channels'.format(
+                                 self.required_channels, image.shape[-1]))
+
+        # Resize image, returns unmodified if appropriate
+        resized_image = self._resize_input(image, image_mpp)
+
+        # Generate model outputs
+        output_images = self._run_model(
+            image=resized_image, batch_size=batch_size,
+            pad_mode=pad_mode, preprocess_kwargs=preprocess_kwargs
+        )
+
+        # Resize output_images back to original resolution if necessary
+        label_image = self._resize_output(output_images, image.shape)
+
+        # Postprocess predictions to create label image
+        predicted_spots = self._postprocess(label_image, **postprocess_kwargs)
+
+        return predicted_spots
+
+    def predict(self,
+                image,
+                batch_size=4,
+                image_mpp=None,
+                pad_mode='reflect',
+                preprocess_kwargs=None,
+                postprocess_kwargs=None,
+                threshold=0.95,
+                clip=False):
+        """Generates a list of coordinate spot locations of the input
+        running prediction with appropriate pre and post processing
+        functions.
+        Input images are required to have 4 dimensions
+        ``[batch, x, y, channel]``.
+        Additional empty dimensions can be added using ``np.expand_dims``.
+        Args:
+            image (numpy.array): Input image with shape
+                ``[batch, x, y, channel]``.
+            batch_size (int): Number of images to predict on per batch.
+            image_mpp (float): Microns per pixel for ``image``.
+            pad_mode (str): The padding mode, one of "constant" or "reflect".
+            preprocess_kwargs (dict): Keyword arguments to pass to the
+                pre-processing function.
+            postprocess_kwargs (dict): Keyword arguments to pass to the
+                post-processing function.
+            threshold (float): Probability threshold for a pixel to be
+                                considered as a spot.
+            clip (bool): Determines if pixel values will be clipped by percentile.
+        Raises:
+            ValueError: Input data must match required rank of the application,
+                calculated as one dimension more (batch dimension) than
+                expected by the model.
+            ValueError: Input data must match required number of channels.
+            ValueError: Threshold value must be between 0 and 1.
+        Returns:
+            numpy.array: Coordinate locations of detected spots.
+        """
+
+        if threshold < 0 or threshold > 1:
+            raise ValueError("""Threshold value must be between 0 and 1.""")
```

Comment:
```suggestion
            raise ValueError('Threshold value must be between 0 and 1.')
```
Generally better to avoid triple quotes outside of docstrings